ckan/views/dataset.py

```diff
@@ -804,8 +804,9 @@ def get(self,
         pkg = context.get(u"package")
         resources_json = h.json.dumps(data.get(u'resources', []))
         user = current_user.name
+        pkg_id = pkg.id if pkg else None
         try:
-            check_access(u'package_update', context)
+            check_access(u'package_update', context, {"id": pkg_id})
```

Comment:
@shashigharti we already have `pkg_dict` object so there should not be any reason to depend on a `package` that comes from the `context` object. (Part of the logic that we are trying to remove.)

Could you clean up this part to call directly to `pkg_dict["id"]` instead of adding a logic to get it from `pkg` object?

Reply:
Thanks! I have removed it.